### PR TITLE
Update Keychain service

### PR DESCRIPTION
**Pull request by vinothvino42**
Ticket: #730 

Added clearKeychain method and update saveToKeychain

### CODE REVIEW

File: lib/framework/data_context.dart [hunk 1]
```diff
@@ -361,12 +361,16 @@ class NativeInvokable with Invokable {
...
+    if (defaultTargetPlatform == TargetPlatform.iOS) {
+      try {
+        const platform = MethodChannel('com.ensembleui.dev/safari-extension');
+        final _ = await platform.invokeMethod('saveToKeychain', {key: value});
+      } on PlatformException catch (e) {
+        throw LanguageError(
+            'Failed to invoke ensemble.saveToKeychain. Reason: ${e.toString()}');
+      }
+    } else {
+      throw LanguageError('ensemble.saveToKeychain is only supported for iOS');
```
Comment (vusters) [inline lib/framework/data_context.dart:373]: What does this mean? This function can now never be used in Ensemble unless you only deploy on iOS?

Comment (vinothvino42) [reply lib/framework/data_context.dart:376]: Now I've done this, especially for iOS and Tin. The method points to safari extension

**Comment by vusters** [comment]
@vinothvino42 see what I wrote above. In our YAML code we simply use `ensemble.saveToKeyChain()`. It should be a no-op on platforms that don't support it. Currently it throws an exception on other platforms

**Comment by vinothvino42** [comment]
> @vinothvino42 see what I wrote above. In our YAML code we simply use `ensemble.saveToKeyChain()`. It should be a no-op on platforms that don't support it. Currently it throws an exception on other platforms

Okay, I'll remove the throwing an exception on other platforms. So now it'll simply work for iOS and nothing will happen for other platforms when using `ensemble.saveToKeychain` or `ensemble.clearKeychain`

**Review by evshi — APPROVED** [review]

File: lib/framework/data_context.dart [hunk 2]
```diff
@@ -361,12 +363,28 @@ class NativeInvokable with Invokable {
   }
 
   Future<void> saveToKeychain(String key, String value) async {
-    try {
-      const platform = MethodChannel('com.ensembleui.dev/safari-extension');
-      final dynamic result =
-          await platform.invokeMethod('saveToKeychain', {key: value});
-    } on PlatformException catch (e) {
-      print('Failed to get value: ${e.message}.');
+    if (defaultTargetPlatform == TargetPlatform.iOS) {
```
[the remark this answers is not in the document]
Comment (vinothvino42) [reply lib/framework/data_context.dart:366]: Okay I'll update now

**Review by vusters — APPROVED** [review]

**Pull request merged**